src/FederationSetup/RecoveryTransactionCreator.cs

```diff
@@ -139,16 +140,51 @@ public FundsRecoveryTransactionModel CreateFundsRecoveryTransaction(bool isSideC
             builder.AddKeys(privateKey);
             builder.AddCoins(coinRefs.Select(c => ScriptCoin.Create(network, c.Transaction.Id, (uint)c.Transaction.Index, c.Transaction.Amount, c.Transaction.ScriptPubKey, oldRedeemScript)));
 
-            // Split the coins into multiple outputs.
             Money amount = coinRefs.Sum(r => r.Transaction.Amount) - fee;
-            const int numberOfSplits = 10;
-            Money splitAmount = new Money((long)amount / numberOfSplits);
-            var recipients = new List<Stratis.Features.FederatedPeg.Wallet.Recipient>();
-            for (int i = 0; i < numberOfSplits; i++)
+
+            if (!burn)
             {
-                Money sendAmount = (i != (numberOfSplits - 1)) ? splitAmount : amount - splitAmount * (numberOfSplits - 1);
+                // Split the coins into multiple outputs.
+                const int numberOfSplits = 10;
+                Money splitAmount = new Money((long)amount / numberOfSplits);
+                Script recipient = redeemScript.PaymentScript;
+
+                for (int i = 0; i < numberOfSplits; i++)
+                {
+                    Money sendAmount = (i != (numberOfSplits - 1)) ? splitAmount : amount - splitAmount * (numberOfSplits - 1);
 
-                builder.Send(redeemScript.PaymentScript, sendAmount);
+                    builder.Send(recipient, sendAmount);
+                }
+            }
+            else
+            {
+                // We don't have the STRAX network classes but we need a STRAX address.
```

Comment:
Isnt it just easier and "safer" to get a strax address pre-generated and pass it to the tool?

Reply:
The address depends on the federation id and that in turn depends on the list of genesis public keys. So its sufficiently complex to require a tool to generate. This might as well be that tool.